pystarport/pystarport/cluster.py

```diff
@@ -328,431 +215,137 @@ def ipport_grpc_tx(self, i):
-        )
+    def gentx(self, name, coins, i=0, min_self_delegation=1):
+        return self.cosmos_cli(i).gentx(name, coins, min_self_delegation)
 
     def collect_gentxs(self, gentx_dir, i=0):
-        return self.raw("collect-gentxs", gentx_dir, home=self.home(i))
+        return self.cosmos_cli(i).collect_gentxs(gentx_dir)
 
     def status(self, i=0):
-        return json.loads(self.raw("status", node=self.node_rpc(i)))
+        return self.cosmos_cli(i).status()
 
     def block_height(self, i=0):
-        return int(self.status(i)["SyncInfo"]["latest_block_height"])
+        return self.cosmos_cli(i).block_height()
 
     def block_time(self, i=0):
-        return isoparse(self.status(i)["SyncInfo"]["latest_block_time"])
+        return self.cosmos_cli(i).block_time()
 
     def balance(self, addr, i=0):
-        coin = json.loads(
-            self.raw(
-                "query", "bank", "balances", addr, output="json", node=self.node_rpc(i)
-            )
-        )["balances"]
-        if len(coin) == 0:
-            return 0
-        coin = coin[0]
-        assert coin["denom"] == "basecro"
-        return int(coin["amount"])
-
-    def distribution_commision(self, addr, i=0):
-        coin = json.loads(
-            self.raw(
-                "query",
-                "distribution",
-                "commission",
-                addr,
-                output="json",
-                node=self.node_rpc(i),
-            )
-        )["commission"][0]
-        return float(coin["amount"])
+        return self.cosmos_cli(i).balance(addr)
+
+    def distribution_commission(self, addr, i=0):
+        return self.cosmos_cli(i).distribution_commission(addr)
 
     def distribution_community(self, i=0):
-        coin = json.loads(
-            self.raw(
-                "query",
-                "distribution",
-                "community-pool",
-                output="json",
-                node=self.node_rpc(i),
-            )
-        )["pool"][0]
-        return float(coin["amount"])
+        return self.cosmos_cli(i).distribution_community()
 
     def distribution_reward(self, delegator_addr, i=0):
-        coin = json.loads(
-            self.raw(
-                "query",
-                "distribution",
-                "rewards",
-                delegator_addr,
-                output="json",
-                node=self.node_rpc(i),
-            )
-        )["total"][0]
-        return float(coin["amount"])
+        return self.cosmos_cli(i).distribution_reward(delegator_addr)
 
     def address(self, name, i=0, bech="acc"):
-        output = self.raw(
-            "keys",
-            "show",
-            name,
-            "-a",
-            home=self.home(i),
-            keyring_backend="test",
-            bech=bech,
-        )
-        return output.strip().decode()
-
-    @format_doc_string(
-        options=",".join(v.value for v in ModuleAccount.__members__.values())
-    )
-    def module_address(self, name):
```

Comment:
this method is put here so it can be called directly in cli:
```
$ pystarport cli - module_address fee_collector
cro17xpfvakm2amg962yls6f84z3kell8c5lgztehv
```

Reply:
Added back, thanks!